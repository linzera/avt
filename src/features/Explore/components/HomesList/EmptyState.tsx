import React from 'react';
import {StyleSheet, View} from 'react-native';
import Typography from '~/components/Typography';
import {useHomeItemHeight} from '@explore/hooks/use-home-item-height';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    maxWidth: 240,
    textAlign: 'center',
  },
  subtitle: {
    maxWidth: 278,
    marginTop: 12,
    textAlign: 'center',
  },
});

export default function EmptyState() {
  const itemHeight = useHomeItemHeight();
  return (
    <View style={[styles.container, {height: itemHeight}]}>
      <Typography
        component="text20"
        fontType="medium"
        color="primary"
        style={styles.title}>
        We couldnâ€™t find any available homes...
      </Typography>
      <Typography color="primary05" style={styles.subtitle}>
        Please, try to select other dates to see available homes inside selected
        regions.
      </Typography>
    </View>
  );
}
