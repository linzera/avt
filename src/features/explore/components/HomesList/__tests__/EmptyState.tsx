import React from 'react';
import {render} from '@testing-library/react-native';
import EmptyState from '../EmptyState';

describe('<EmptyState />', () => {
  it('should render properly', () => {
    const {toJSON} = render(<EmptyState />);
    expect(toJSON()).toMatchSnapshot();
  });
});
