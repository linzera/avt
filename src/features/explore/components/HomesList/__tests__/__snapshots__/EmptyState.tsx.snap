// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState /> should render properly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "height": 1120.56,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#022B54",
        "fontFamily": "SourceSansPro-SemiBold",
        "fontSize": 20,
        "maxWidth": 240,
        "textAlign": "center",
      }
    }
  >
    We couldnâ€™t find any available homes...
  </Text>
  <Text
    style={
      Object {
        "color": "#022B5450",
        "fontFamily": "SourceSansPro-Regular",
        "fontSize": 16,
        "marginTop": 12,
        "maxWidth": 278,
        "textAlign": "center",
      }
    }
  >
    Please, try to select other dates to see available homes inside selected regions.
  </Text>
</View>
`;
